crates/bevy_window/src/window.rs

```diff
@@ -86,36 +96,103 @@ pub enum CompositeAlphaMode {
 }
 
-impl WindowId {
-    /// Creates a new [`WindowId`].
-    pub fn new() -> Self {
-        WindowId(Uuid::new_v4())
-    }
-    /// The [`WindowId`] for the primary window.
-    pub const fn primary() -> Self {
-        WindowId(Uuid::from_u128(0))
-    }
-    /// Get whether or not this [`WindowId`] is for the primary window.
-    pub fn is_primary(&self) -> bool {
-        *self == WindowId::primary()
-    }
+/// Defines the way a window is displayed
+#[derive(Default, Debug, Component, Clone, Copy, PartialEq, Reflect)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[reflect(Component)]
+pub enum WindowMode {
+    /// Creates a window that uses the given size.
+    #[default]
+    Windowed,
+    /// Creates a borderless window that uses the full size of the screen.
+    BorderlessFullscreen,
+    /// Creates a fullscreen window that will render at desktop resolution. The app will use the closest supported size
+    /// from the given size and scale it to fit the screen.
+    SizedFullscreen,
+    /// Creates a fullscreen window that uses the maximum supported size.
+    Fullscreen,
 }
 
-use crate::CursorIcon;
-use std::fmt;
-
-use crate::raw_handle::RawHandleWrapper;
+/// Define how a window will be created and how it will behave.
+#[derive(Default, Bundle, Debug, Clone)]
+pub struct WindowBundle {
+    /// Marker component denoting the entity as a window.
+    pub window: Window,
+    /// The cursor of this window.
+    pub cursor: Cursor,
+    /// The position of this window's cursor.
+    pub cursor_position: CursorPosition,
+    /// What presentation mode to give the window.
+    pub present_mode: PresentMode,
+    /// Which fullscreen or windowing mode should be used?
+    pub mode: WindowMode,
+    /// Where the window should be placed.
+    pub position: WindowPosition,
+    /// What resolution the window should have.
+    pub resolution: WindowResolution,
+    /// Stores the title of the window.
+    pub title: WindowTitle,
+    /// Should the window start minimized, maximized, normal?
+    pub state: WindowState,
+    /// How the alpha channel of textures should be handled while compositing.
+    pub composite_alpha_mode: CompositeAlphaMode,
+    /// Which size limits to give the window.
+    pub resize_constraints: WindowResizeConstraints,
+    /// Should the window be resizable?
+    pub resizable: WindowResizable,
+    /// Should the window have decorations enabled?
+    pub decorations: WindowDecorations,
+    /// Should the window be transparent?
+    pub transparency: WindowTransparency,
+    /// Should the window start focused?
+    pub focus: WindowFocus,
+    /// How the window will interact if it is on a web platform rather than OS window.
+    pub canvas: WindowCanvas,
+}
 
-impl fmt::Display for WindowId {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.as_simple().fmt(f)
-    }
+#[derive(WorldQuery)]
+pub struct WindowComponents<'a> {
```

Comment:
We should be using `'static` lifetimes here to clean this up: these are just marker lifetimes, and aren't actually used.